### DISCUSSION TITLE
have any ideas that support the web?

### DISCUSSION THREAD

**Discussion by Jzow** [modular/modular#1252] [General]
We are currently using `Java` and `Kotlin` as our backend web language for business development. The front-end uses `TypeScript` and` Vue`. With the development of our business, we need to use some deep learning technologies. Our team has been considering using Java for deep learning tasks or Rust for a long time.

We have conducted some experiments, and we have been using `CoreNLP` from Stanford and `DJL` from AWS before. However, the team believes that its performance in computing is still inferior to `Python` and `Rust`. The Java project is deployed on Linux, but the JVM has always been our concern. Recently, we saw news that `Java22` external function API can directly access functions in memory

We have long been looking for a reasonable architecture that meets both our existing web application services and our algorithmic needs at minimal cost.

The development of a language cannot be separated from the ecological construction of the community, and currently Rust ecology is insufficient in deep learning.

Just as both Rust and Python can support the Web, my question is whether `Mojo` will consider supporting Http-related packages

**Comment by Sharktheone** [comment]
You could just import a Python web library and  for better performance you could use ffi and program the web stuff in rust. I think you can either compile the rust side to a staticlib and embed it directly in mojo. The other option could be to compile the mojo side as shared lib (i havn't seen this in the documentation but someone doing this on github) and call the needed functions from the rust side.

**Reply by manatlan** [reply]
> The other option could be to compile the mojo side as shared lib (i havn't seen this in the documentation but someone doing this on github) and call the needed functions from the rust side.

Have you got any pointers ?

**Reply by Sharktheone** [reply]
It's with C but it should work with rust: https://github.com/ihnorton/mojo-ffi

**Reply by Jzow** [reply]
In short, is it through mojo to call my api in rust web ?

**Reply by Sharktheone** [reply]
This example just shows how to use ffi with mojo, but you can use it to integrate mojo in a rust project / rust in a mojo project.
